src/userdb/userwork.c

```diff
@@ -80,12 +80,14 @@ static int build_user_json(Varlink *link, UserRecord *ur, JsonVariant **ret) {
         assert(ur);
         assert(ret);
 
-        r = varlink_get_peer_uid(link, &peer_uid);
+        r = varlink_get_uid(link, &peer_uid);
         if (r < 0) {
-                log_debug_errno(r, "Unable to query peer UID, ignoring: %m");
+                log_debug_errno(r, "Unable to query requestor's UID, ignoring: %m");
                 trusted = false;
-        } else
+        } else {
                 trusted = peer_uid == 0 || peer_uid == ur->uid;
+                log_debug("Peer UID is " UID_FMT "; %strusted", peer_uid, (trusted ? "" : "un"));
+        }
```

Comment:
you might wonder why SO_PEERCRED is safer to rely on than SCM_CREDENTIALS? that's because it's easy to get a suid program to write something to an fd of your choice. It's much harder to get it to call connect() on an fd of your choice, because that's what pins the peer cred.

(this is the reason why dbus for example also prefers using SO_PEERCRED rather than SCM_CREDENTIALS)